src/App.tsx

```diff
@@ -5,7 +5,20 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = (movie: Movie) => {
+    const hasMovie = movies
+      .some(addedMovie => addedMovie.imdbId === movie.imdbId);
+
+    if (hasMovie) {
+      return;
+    }
+
+    const updatedMovies = [...movies, movie];
+
+    setMovies(updatedMovies);
```

Comment:
```suggestion
   

    setMovies(prevMovies => [...prevMovies, movie]);
```
It is good practice to pass callback to your setter function if new state value depends on its previous value

Reply:
Implemented. Thank you for the review!